akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
+            Behaviors.same
+          case (ctx, UnWatch(ref)) ⇒
+            ctx.unwatch(ref)
+            probe.ref ! Pong
+            Behaviors.same
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
+      )
+      actor ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      actor ! UnWatch(childRef)
+      probe.expectMessage(Pong)
+      actor ! Watch(childRef)
+      probe.expectMessage(Pong)
+      childRef ! Stop
+      probe.expectMessage(GotSignal(Terminated(childRef)(null)))
     }
 
     "terminate upon not handling Terminated" in {
-      sync(setup("ctx13", ignorePostStop = false) { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith.mkChild(None, ctx.spawnMessageAdapter(ChildEvent), self).keep {
-          case (subj, child) ⇒
-            muteExpectedException[DeathPactException]()
-            subj ! Watch(child, self)
-        }.expectMessageKeep(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(Watched)
-            subj ! BecomeCareless(self)
-        }.expectMessageKeep(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(BecameCareless)
-            child ! Stop
-        }.expectMessage(expectTimeout) {
-          case (msg, (subj, child)) ⇒
-            msg should ===(ChildEvent(GotSignal(PostStop)))
-        }.expectMessage(expectTimeout) {
-          case (msg, _) ⇒
-            msg should ===(GotSignal(PostStop))
-        }
-      })
+      val probe = TestProbe[Event]()
+      val child = (Behaviors.immutablePartial[Command] {
+        case (_, Stop) ⇒
+          Behaviors.stopped
+      } onSignal {
+        case (_, signal) ⇒
+          probe.ref ! GotChildSignal(signal)
+          Behavior.same
+      }).decorate
+      val actor = spawn(
+        (Behaviors.immutablePartial[Command] {
+          case (ctx, MakeChild) ⇒
+            val childRef = ctx.spawn(child, "A")
+            ctx.watch(childRef)
+            probe.ref ! ChildMade(childRef)
+            Behaviors.same
+          case (_, Inert) ⇒
+            probe.ref ! InertEvent
+            Behaviors.immutable[Command] {
+              case (_, _) ⇒ Behaviors.unhandled
+            } onSignal {
+              case (_, Terminated(_)) ⇒ Behaviors.unhandled
+              case (_, signal) ⇒
+                probe.ref ! GotSignal(signal)
+                Behaviors.same
+            }
+        } onSignal {
+          case (_, signal) ⇒
+            probe.ref ! GotSignal(signal)
+            Behaviors.same
+        }).decorate
+      )
+      actor ! MakeChild
+      val childRef = probe.expectMessageType[ChildMade].ref
+      actor ! Inert
+      probe.expectMessage(InertEvent)
+      childRef ! Stop
+      probe.expectMessage(GotChildSignal(PostStop))
+      probe.expectMessage(GotSignal(PostStop))
```

Comment:
This is not right, it should test that if the actor watches the child, but does not handle the `Terminated` message, it is itself stopped. Again `probe.expectTerminated` is your friend, to see that `actor` terminates when not handling `Terminated`

Reply:
What about this comment?

Reply:
Was addressed: used `probe.expectTerminated` to verify that parent actor is stopped after reacting on child's termination in the following manner: `case (_, Terminated(_)) ⇒ Behaviors.unhandled`

Reply:
Good, thanks!